packages/format-currency/src/index.ts

```diff
@@ -1,365 +1,129 @@
+ */
+async function geolocateCurrencySymbol(): Promise< void > {
+	const geoData = await globalThis
+		.fetch?.( GEO_LOCATION_ENDPOINT_URL )
+		.then( ( response ) => response.json() )
+		.catch( ( error ) => {
+			// Do nothing if the fetch fails.
 			// eslint-disable-next-line no-console
-			console.warn(
-				`getCurrencyObject was called with a non-existent currency "${ code }"; falling back to ${ fallbackCurrency }`
-			);
-			return fallbackCurrency;
-		}
-		return code;
-	}
-
-	function getCurrencyOverride( code: string ): CurrencyOverride | undefined {
-		if ( code === 'USD' && geoLocation !== '' && geoLocation !== 'US' ) {
-			return { symbol: 'US$' };
-		}
-		return defaultCurrencyOverrides[ code ];
-	}
+			console.warn( 'Fetching geolocation for format-currency failed.', error );
+		} );
 
-	function doesCurrencyExist( code: string ): boolean {
-		return Boolean( getCurrencyOverride( code ) );
-	}
+	geoLocation =
+		'string' === typeof ( geoData as WithGeoCountry )?.country_short ? geoData.country_short : '';
+}
 
-	/**
-	 * Set a default locale for use by `formatCurrency` and `getCurrencyObject`.
-	 *
-	 * Note that this is global and will override any browser locale that is set!
-	 * Use it with care.
-	 */
-	function setDefaultLocale( locale: string | undefined ): void {
-		defaultLocale = locale;
+/**
+ * Retrieves the currency override for a given currency.
+ * If the currency is USD and the user is not in the US, it will return `US$`.
+ */
+function getCurrencyOverride( currency: string ): CurrencyOverride | undefined {
+	if ( currency === 'USD' && geoLocation !== '' && geoLocation !== 'US' ) {
+		return { symbol: 'US$' };
 	}
+	return defaultCurrencyOverrides[ currency ];
+}
 
-	function getPrecisionForLocaleAndCurrency(
-		locale: string,
-		currency: string
-	): number | undefined {
-		const formatter = getFormatter( 0, currency, { locale } );
-		return formatter.resolvedOptions().maximumFractionDigits ?? 3; // 3 is the default for Intl.NumberFormat if minimumFractionDigits is not set
+/**
+ * Returns a valid currency code based on a shortlist of currency codes.
+ * Only currencies from the shortlist are allowed. Everything else will fall back to `FALLBACK_CURRENCY`.
+ */
+function getValidCurrency( currency: string ): string {
+	if ( ! getCurrencyOverride( currency ) ) {
+		// eslint-disable-next-line no-console
+		console.warn(
+			`getCurrencyObject was called with a non-existent currency "${ currency }"; falling back to ${ FALLBACK_CURRENCY }`
+		);
+		return FALLBACK_CURRENCY;
 	}
-
-	return {
-		formatCurrency,
-		getCurrencyObject,
-		setDefaultLocale,
-		geolocateCurrencySymbol,
-	};
+	return currency;
 }
 
 function getLocaleFromBrowser() {
 	if ( typeof window === 'undefined' ) {
-		return fallbackLocale;
+		return;
 	}
 	if ( window.navigator?.languages?.length > 0 ) {
 		return window.navigator.languages[ 0 ];
 	}
-	return window.navigator?.language ?? fallbackLocale;
+	return window.navigator?.language;
```

Comment:
This is just so we don't mix concerns. `getLocaleFromBrowser` returning `undefined` is a valid state when none is defined. We inject the fallback from `getLocaleToUse`.